### PR TITLE
test: Use MiniWallet in mempool_persist

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

File: test/functional/mempool_persist.py [hunk 1]
```diff
@@ -111,17 +123,19 @@ def run_test(self):
...
+            self.nodes[2].loadwallet("watch")
+            wallet_watch = self.nodes[2].get_wallet_rpc("watch")
+            self.nodes[2].syncwithvalidationinterfacequeue()  # Flush mempool to wallet
+            assert_equal(node2_balance, wallet_watch.getbalance())
 
-        # start node0 with wallet disabled so wallet transactions don't get resubmitted
         self.log.debug("Stop-start node0 with -persistmempool=0. Verify that it doesn't load its mempool.dat file.")
         self.stop_nodes()
-        self.start_node(0, extra_args=["-persistmempool=0", "-disablewallet"])
+        self.start_node(0, extra_args=["-persistmempool=0"])
```
Comment (Shubhankar-Gambhir) [inline test/functional/mempool_persist.py:138]: shouldn't node be started with disabled wallet?

Comment (Shubhankar-Gambhir) [reply test/functional/mempool_persist.py:138]: Feel free to ignore this but you can use self.restart_node() here.

Comment (MarcoFalke) [reply test/functional/mempool_persist.py:138]: No wallet (that has any unconfirmed txs) should exist for node 0

**Comment by laanwj** [comment]
Code review ACK faae0988d6abb50039026a49543eef134eb9103f

**Pull request merged**